config/oauth_visitors.go

```diff
@@ -56,3 +56,12 @@ func refreshableVisitor(inner oauth2.TokenSource) func(r *http.Request) error {
 		return nil
 	}
 }
+
+func azureVisitor(workspaceResourceId string, inner func(*http.Request) error) func(*http.Request) error {
+	return func(r *http.Request) error {
+		if workspaceResourceId != "" {
+			r.Header.Set("X-Databricks-Azure-Workspace-Resource-Id", workspaceResourceId)
```

Comment:
Besides this header, we may need to add service management header as well: https://github.com/databricks/terraform-provider-databricks/blob/v1.9.2/common/azure_auth.go#L147

Reply:
Isn't that already added by the `serviceToServiceVisitor`?

The header in question: `X-Databricks-Azure-SP-Management-Token`.

Reply:
After looking at this original change, we realized that there was a secondary problem: if a user logs in via the CLI using a service principal, the CLI auth type needs to also request the management token from the CLI in addition to the Databricks token. The point is: all Azure-native auth types need to call `azureVisitor`, and all auth types that need to include a second token in the request need to call `serviceToServiceVisitor`; this now includes the CLI.